src/tests/ftest/util/fio_utils.py

```diff
@@ -68,9 +75,147 @@ def __init__(self, namespace, command, test):
         self.trigger_remote = FormattedParameter("--trigger-remote={}")
         self.aux_path = FormattedParameter("--aux-path={}")
 
-        # fio job options
+        # Middleware to use with fio.  Needs to be configured externally prior
+        # to calling run().
+        self.api = BasicParameter(None, "POSIX")
+
+        # List of fio job names to run
+        self.names = BasicParameter(None)
+        self._jobs = {}
+
+        # List of hosts on which the fio command will run.  If not defined the
+        # fio command will run locally
+        self._hosts = None
+
+    @property
+    def hosts(self):
+        """Get the host(s) on which to remotely run the fio command via run().
+
+        Returns:
+            list: remote host(s) on which the fio command will run.
+
+        """
+        return self._hosts
+
+    @hosts.setter
+    def hosts(self, value):
+        """Set the host(s) on which to remotely run the fio command via run().
+
+        If the specified host is None the command will run locally w/o ssh.
+
+        Args:
+            value (list): remote host(s) on which to run the fio command
+        """
+        if value is None or isinstance(value, list):
+            self._hosts = value
+        else:
+            self.log.error("Invalid fio host list: %s (%s)", value, type(value))
+            self._hosts = None
+
+    def get_params(self, test):
+        """Get values for all of the command params from the yaml file.
+
+        Args:
+            test (Test): avocado Test object
+        """
+        super(FioCommand, self).get_params(test)
+
+        # Add jobs
+        self._jobs.clear()
+        if self.names.value is not None:
+            for name in self.names.value:
+                self._jobs[name] = FioJob(name)
```

Comment:
(pylint-no-value-for-parameter) No value for argument 'name' in constructor call

Reply:
Fixed